src/test/java/com/google/api/codegen/testdata/java_mock_service_impl_library.baseline

```diff
@@ -297,8 +284,7 @@ public class MockLibraryServiceImpl implements LibraryService  {
   public void updateBookIndex(UpdateBookIndexRequest request,
       StreamObserver<Empty> responseObserver) {
     if (responses != null && !responses.isEmpty()) {
-      Empty response = (Empty) responses.get(0);
-      responses.remove(0);
+      Empty response = (Empty) responses.poll();
```

Comment:
Depends on whether it's due to a bug in our code (throwing an exception and breaking the flow of execution makes that clear), or a bug in the user test configuration (for which the graceful `onError()` mechanism is appropriate).

I like breaking hard and fast when we have a bug, unless it's a mission critical system like a web app. Even back when I worked in the Linux kernel, I saw that `panic()` was often used whenever data structures were found to be in a surprising state, to ensure that the underlying bug got fixed.